microceph/ceph/config.go

```diff
@@ -3,15 +3,140 @@ package ceph
 import (
 	"context"
 	"database/sql"
+	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"github.com/canonical/microceph/microceph/api/types"
 	"github.com/canonical/microceph/microceph/common"
 	"github.com/canonical/microceph/microceph/database"
 )
 
+type ConfigExtras struct{
+	Who     string // Ceph Config internal <who> against each key
+	Daemons []string // Daemons that need to be restarted post config change.
+	Regexp  string // Regular Expression to check value against key
+	Cb      *func(s common.StateInterface, c types.Config)
```

Comment:
I believe neither Regexp nor Cb are currently in use, right? Suggest removing them if so

Reply:
Ack